### ISSUE TITLE
malicious code found !?

### ISSUE THREAD

**Issue by Zwilla** [bitmaintech/bmminer#7]
No one is able to connect to you service at: 
auth.minerlink.com

#define AUTH_URL    "auth.minerlink.com"  
static bool setup_send_mac_socket(char * s)

**Comment by lichtamberg** [comment]
http://www.antbleed.com/

**Comment by Merthurian** [comment]
And the quarterback is TOAST!

**Comment by wodim** [comment]
friendly reminder all ancaps must die

**Comment by n4ru** [comment]
Shitstorm incoming!

**Comment by dhimmel** [comment]
The shut down code is at [`driver-btm-c5.c#L7755-L7756`](https://github.com/bitmaintech/bmminer/blob/b5de92908498590d96d333a1e2570eab0eb321d3/driver-btm-c5.c#L7755-L7756):

```c
        if(strstr(rec,"false"))
            if_stop = true;
```

**Comment by Zwilla** [comment]
I love: [http://www.antbleed.com/](https://www.zwilla.de/goods/english-software/donation-product-for-changing-bmminer-into-cgminer-4-9-2/?lang=en)

**Comment by Zorlin** [comment]
GG Bitmain

**Comment by Fazio** [comment]
First of all, I apologize for this. I uploaded some uncompleted feature' code and caused considerable misunderstandings.
This uncompleted feature was intended to allow the owners of Antminer to remotely shut down their miners that may have been stolen or hijacked by their hosting service provider. We never intended to use this feature on any Antminer without authorization from its owner. This is similar to the remote erase or shutdown feature provided by most famous smartphone manufacturers.
However, this feature was never completed. But I made a mistake to upload the test code to github without checking carefully, I'm really apologize for this.
The controversy around this code has brought our attention to improve the design in order to address vulnerabilities that were pointed out by the community recently. For this, we thank the community for pointing out bugs and contributing to our open-source codebase. We would also like to take this opportunity to express our solidified belief in the open source community and express our renewed commitment to improving the quality and testing methods of our code. 
Moving on, we have released the new updated source-code on GitHub and new firmware on our website which removes this bug. We recommend all Antminer owners to upgrade their firmware to the updated firmware from the list below and strongly advise to not download any firmware provided by a third-party which can lead to unexpected outcomes in function and can also be potential phishing attacks. 

S9： 
https://shop.bitmain.com/files/download/Antminer-S9-all-201704270135-autofreq-user-Update2UBI-NF.tar.gz
T9：
https://shop.bitmain.com/files/download/Antminer-T9-all-201704270140-autofreq-user-Update2UBI-NF.tar.gz
R4：
https://shop.bitmain.com/files/download/Antminer-R4-all-201704270142-autofreq-user-Update2UBI-NF.tar.gz
L3：
https://shop.bitmain.com/files/download/Antminer-L3-201704271449-384M.tar.gz
L3+：
https://shop.bitmain.com/files/download/Antminer-L3+-201704271426-384M.tar.gz

Source-code for Antminer S9, T9 and R4:
git@github.com/bitmaintech/bmminer-mix.git
Source-code for Antminer L3 and L3+:
git@github.com/bitmaintech/setup-A8.git which includes (git@github.com/bitmaintech/ltc_frimware.git) inside.

This project is currently  not maintained any more.  please see git@github.com:bitmaintech/bmminer-mix.git.

**Comment by seweso** [comment]
This issue really cannot be super important and egregious that it suddenly needs it's own website to promote the issue. While businesses fail to firewall their operation, protect against DNS attacks or check even for open issues to keep their mining operation secure. 

Tempest in a teapot, again is what it seems like to me. Especially considering stratum already being equally vulnerable, while giving an attacker WAY more control. The outrage about this issue is coming from another source of anger. 

The thing to learn is: check your issue tracker. ;)

**Comment by shesek** [comment]
> Especially considering stratum already being equally vulnerable, while giving an attacker WAY more control. 

The stratum server is configurable by the user, this is not.

**Comment by ftrader** [comment]
Impressed by how quickly you issued updates to fix this, and issuing a professional company response within a day. Good work.

**Comment by Cocosoft** [comment]
@ftrader Well this time they had no one to slander compared to the AsicBoost response.

**Comment by Mirobit** [comment]
@ftrader the fix is removing 3 lines of code. Not sure if it is impressiv to do this within 2 days.

I get that some people don't like core but this fanboyism of bitmain is scary.

**Comment by n4ru** [comment]
@seweso An estimated 70% of a network with a $12 Billion market cap being shut down by a single DNS hijack is "a tempest in a teapot"? You're kidding, right?

**Comment by ftrader** [comment]
@n4ru : please do provide sources for your "70%" - remember these must be connected to the Internet for your hypothetical DNS hijack.

If you don't provide sources, please think about whether gross exaggerations are helpful.

**Comment by n4ru** [comment]
@ftrader The 12 was a typo which I corrected. I said estimated 70%. AntMiner is the most sold ASIC miner, it is indisputable that it has the most market share. That alone is enough cause for concern. Minimizing or deflecting what happened here doesn't help.

**Comment by Joo5t** [comment]
omg this is serious....antpool can use a DDOS style attack on other blockchains with there sold devices... http://www.antbleed.com/  Is this deliberate backdoor without informing there customers even legal? Can Antpool be sued over this? Will they be sued?

**Comment by t3kk3n** [comment]
The antbleed website is proof that people are quick to jump the gun. Stop!

**Comment by lichtamberg** [comment]
Yeah quick. Did you see when this issue was opened? more than a half year ago.
